product_list_price_from_pricelist/models/product_pricelist.py

```diff
@@ -0,0 +1,125 @@
+                        "Its should not have items with minimum quantity, "
+                        "please check the pricelist %(pricelist)s with item %(name)s."
+                    )
+                    % {
+                        "pricelist": item.pricelist_id.name,
+                        "name": item.name or item.id,
+                    }
+                )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_pricelist_items = super().create(vals_list)
+        test_condition = not config["test_enable"] or (
+            config["test_enable"]
+            and self.env.context.get("test_compute_list_price_from_pricelist")
+        )
+        # If any field from the expected ones is changed,
+        # recompute the product's sales price.
+        if test_condition:
+            for pricelist in new_pricelist_items.pricelist_id:
+                used_in_company = bool(
+                    self.env["res.company"].search_count(
+                        [("base_pricelist_compute_price_id", "=", pricelist.id)]
+                    )
+                )
+                if used_in_company:
+                    pricelist._update_product_price_from_pricelist(new_pricelist_items)
+        return new_pricelist_items
+
+    def write(self, vals):
+        res = super().write(vals)
+        test_condition = not config["test_enable"] or (
+            config["test_enable"]
+            and self.env.context.get("test_compute_list_price_from_pricelist")
+        )
+        # If any field from the expected ones is changed,
+        # recompute the product's sales price.
+        if test_condition and set(vals.keys()).intersection(
+            self._get_fields_trigger_to_recompute_sale_price()
+        ):
+            for pricelist in self.pricelist_id:
+                used_in_company = bool(
+                    self.env["res.company"].search_count(
+                        [("base_pricelist_compute_price_id", "=", pricelist.id)]
+                    )
+                )
+                if used_in_company:
+                    pricelist._update_product_price_from_pricelist(self)
+        return res
+
+    @api.model
+    def _get_fields_trigger_to_recompute_sale_price(self):
+        fields_triggers = [
+            "applied_on",
+            "base",
+            "base_pricelist_id",
+            "categ_id",
+            "compute_price",
+            "date_start",
+            "date_end",
+            "fixed_price",
+            "percent_price",
+            "price_discount",
+            "price_surcharge",
+            "price_round",
+            "product_tmpl_id",
+            "product_id",
+        ]
+        return fields_triggers
+
+    def _get_all_products_from_pricelist_item(self):
+        self.ensure_one()
+        products = self.env["product.template"]
+        if self.applied_on == "3_global":
+            products = self.env["product.template"].search([])
```

Comment:
If you are returning templates, then all the nomenclature and upper variables should be changed from `product` to `template`.